layouts/shortcodes/feature-gate-table.html

```diff
@@ -0,0 +1,114 @@
+                {{- range $stage := $featureGateFile.Params.stages -}}
+                    {{- if or (eq ($stage).stage "stable") (eq ($stage).stage "deprecated") -}}
+                        {{- $graduatedOrDeprecatedFlag = true -}}
+                    {{- end -}}
+                {{- end -}}
+
+                {{- if eq $onlyDisplayAlphaBetaFeature $graduatedOrDeprecatedFlag -}}
+                    {{- continue -}}
+                {{- end -}}
+
+            {{- else -}}
+            <!-- Check if 'removed' parameter is not specified, continue to the next iteration -->
+            {{- if not $removedStatusForFeature -}}
+                {{- continue -}}
+            {{- end -}}
+        {{- end -}}
+
+        {{- range $featureGate := $featureGateFile.Params.stages -}}
+            <!-- Check if the 'stage' value is valid -->
+            {{- $validStages := slice "alpha" "beta" "stable" "deprecated" -}}
+            {{- $isValidStage := in $validStages $featureGate.stage -}}
+            {{- if not $isValidStage -}}
+                {{- errorf "Invalid 'stage' value '%s' specified for feature gate '%s' in %s (Valid values are: %s)" $featureGate.stage $featureGateName (print $featureGateFile.File.Lang "/" $featureGateFile.File.Path) $validStages -}}
+            {{- end -}}
+
+            <!-- Parse and extract the numeric values in format x.y or x.y.z for comparison -->
+            {{- $formattedFromVersion := printf "%s" $featureGate.fromVersion | replaceRE "(\\d+\\.\\d+(?:\\.\\d+)?)?.*" "$1" -}}
+            {{- $formattedToVersion := printf "%s" $featureGate.toVersion | replaceRE "(\\d+\\.\\d+(?:\\.\\d+)?)?.*" "$1" -}}
+
+            <!-- Check for invalid 'fromVersion' value -->
+            {{- if not (eq (printf "%s" $featureGate.fromVersion) $formattedFromVersion) -}}
+                {{- errorf "Invalid 'fromVersion' value '%s' specified for feature gate '%s' in %s" $featureGate.fromVersion $featureGateName (print $featureGateFile.File.Lang "/" $featureGateFile.File.Path) -}}
+            {{- end -}}
+
+            <!-- Check for invalid 'toVersion' value -->
+            {{- if and $featureGate.toVersion (not (eq (printf "%s" $featureGate.toVersion) $formattedToVersion)) -}}
+                {{- errorf "Invalid 'toVersion' value '%s' specified for feature gate '%s' in %s" $featureGate.toVersion $featureGateName (print $featureGateFile.File.Lang "/" $featureGateFile.File.Path) -}}
+            {{- end -}}
+
+            <!-- Display feature gate information in table rows -->
+            <tr>
+                <td><code>{{- $featureGateName -}}</code></td>
+                <td>{{- if isSet $featureGate "defaultValue" -}}<code>{{- $featureGate.defaultValue -}}</code>{{- else -}}–{{- end -}}</td>
+                <td>{{- if eq $featureGate.stage "stable" -}}GA{{- else -}}{{- $featureGate.stage | title -}}{{- end -}}</td>
```

Comment:
We should localize the strings (“Alpha”, “Beta”, “GA”, “Deprecated”) rather than just converting some to title case and substituting GA for stable.

_Tip_: Use HTML, so that we can have `<abbr title="Generally available">GA</abbr>` as the English text for (eg) `feature_stage_stable`.